src/ralph/backends/data/clickhouse.py

```diff
@@ -0,0 +1,401 @@
+                username=self.settings.USERNAME,
+                password=self.settings.PASSWORD,
+                settings=self.settings.CLIENT_OPTIONS.dict(),
+            )
+        return self._client
+
+    def status(self) -> DataBackendStatus:
+        """Check ClickHouse connection status.
+
+        Returns:
+            DataBackendStatus: The status of the data backend.
+        """
+        try:
+            self.client.query("SELECT 1")
+        except ClickHouseError:
+            return DataBackendStatus.AWAY
+
+        return DataBackendStatus.OK
+
+    def list(
+        self, target: str = None, details: bool = False, new: bool = False
+    ) -> Iterator[Union[str, dict]]:
+        """List documents for a given table.
+
+        Args:
+            target (str): The table to list documents from.
+            details (bool): Get detailed archive information instead of just ids.
+            new (bool): Given the history, list only not already fetched archives.
+        """
+        sql = "SHOW TABLES"
+        self.client.database = target if target else self.database
+        for document in self.client.query(sql).named_results():
+            if details:
+                yield document
+            else:
+                yield str(document.get("name"))
+
+    @enforce_query_checks
+    def read(
+        self,
+        *,
+        query: Union[str, ClickHouseQuery] = None,
+        target: str = None,
+        chunk_size: Union[None, int] = 500,
+        raw_output: bool = False,
+        ignore_errors: bool = False,
+    ) -> Iterator[Union[bytes, dict]]:
+        """Read documents matching the query in the target table and yield them.
+
+        Args:
+            query (str or ClickHouseQuery): The query to use when fetching documents.
+            target (str or None): The target table name to query.
+                If target is `None`, the `event_table_name` is used instead.
+            chunk_size (int or None): The chunk size for reading batches of documents.
+                If chunk_size is `None` it defaults to 500.
+            raw_output (bool): Controls whether to yield dictionaries or bytes.
+            ignore_errors (bool): Ignored.
+
+        Yields:
+            bytes: The next raw document if `raw_output` is True.
+            dict: The next JSON parsed document if `raw_output` is False.
+
+        Raises:
+            BackendException: If a failure occurs during ClickHouse connection.
+        """
+        reader = self._read_raw if raw_output else self._read_dict
+        fields = ",".join(query.return_fields) if query.return_fields else "event"
+
+        if target is None:
+            target = self.event_table_name
+        sql = f"SELECT {fields} FROM {target}"  # nosec
+
+        if query.where_clause:
+            sql += f"  WHERE {query.where_clause}"
+
+        if chunk_size is None:
+            chunk_size = self.default_chunk_size
+
+        sql += f" LIMIT {chunk_size}"
```

Comment:
You're right!
Now using `chunk_size` uses `buffer_size` and I added the `limit` query parameter.
Also changed `ClickHouseLRSBackend` to use `ClickHouseDataBackend.read` method instead of `_find`